nestkernel/conn_builder.cpp

```diff
@@ -382,9 +382,18 @@ nest::ConnBuilder::change_connected_synaptic_elements( index sgid,
 void
 nest::ConnBuilder::connect()
 {
+  if ( kernel().model_manager.connector_requires_symmetric( synapse_model_ )
+    && not symmetric_ && not is_symmetric() )
```

Comment:
@janhahne @jougs This looks good, although I think that we could simplify the method to
```C++
inline bool
is_symmetric() const
{
  return sources == targets && all_parameters_scalar();
}
```
The other method would be something like
```C++
bool
ConnBuilder::all_parameters_scalar_() const
{
  bool all_scalar = weight_->is_scalar() && delay_->is_scalar();
  for ( ConnParameterMap::const_iterator it = synapse_params_.begin() ; ... ; ++it )
  {
     all_scalar = all_scalar && it->second->is_scalar();
  }
  return all_scalar;
}
```
We then need in `ConnParameter` and extra method `is_scalar()` that returns true only for `Scalar{Double,Integer}Parameter`.  We cannot use existing methods since `is_array()` returns false for random numbers and `number_of_values()` returns 0 for scalar and random.